### PR TITLE
Contracts interface expansion

### PR DESCRIPTION
**Pull request by jacogr**
0. Modify `contracts/Registry` to have `lookupMeta` (interface to contract `get`) method
0. Fixed `contracts/GithubHint` to retrieve instance properly & add `getEntry` to retrieve a specific entry

(Split from https://github.com/ethcore/parity/pull/4178)

### CODE REVIEW

File: js/src/contracts/githubhint.js [hunk 1]
```diff
@@ -27,6 +28,19 @@ export default class GithubHint {
...
-    return this.getContract().instance;
+    if (this._instance) {
+      return Promise.resolve(this._instance);
+    }
+
+    return this.getContract().then((contract) => {
+      this._instance = contract.instance;
+      return this._instance;
+    });
+  }
```
Comment (jacogr) [inline js/src/contracts/githubhint.js]: Implementation aligns with what we have in the others in `~/contracts` now - previously it actually didn't work at all.

File: js/src/contracts/registry.js [hunk 2]
```diff
@@ -101,4 +101,17 @@ export default class Registry {
       return address;
     });
   }
+
+  lookupMeta (_name, key) {
```
Comment (ngotchac) [inline js/src/contracts/registry.js]: Wouldn't it make sense to now use this `lookupMeta` method in `lookupAddress` instead of having twice the same code?

Comment (jacogr) [inline js/src/contracts/registry.js]: I had it that way around initially since it is 99.9% shared. Not sure why it was reverted, will take a re-look.

Comment (jacogr) [inline js/src/contracts/registry.js]: I see why... one calls `get` the other `getAddress` - so could be clever and obfuscate the code to pass the key as well as the relevant method to some helper or duplicate with a single-line difference for clarity. 

I opted for the latter.

Comment (ngotchac) [inline js/src/contracts/registry.js]: Ah yes good call. Well I guess that's fine, it small enough. Could have a

```
getLookupParams (_name, key) {
  const name = _name.toLowerCase();
  const sha3 = this._api.util.sha3.text(name);

  return [sha3, key];
}
```

method though

**Pull request merged**